### ISSUE TITLE
 elpy and cygwin

### ISSUE THREAD

**Issue by benrifkind** [jorgenschaefer/elpy#921]
I had a similar issue as [518](https://github.com/jorgenschaefer/elpy/issues/518). 
I'm on Windows with Cygwin. My python distribution is a windows distribution and emacs was installed through cygwin. I think there is a misunderstanding of paths between python and Cygwin. Python expects Windows path and Cygwin has Unix paths.
Eg., the initial path to elpy is   `"/home/user/.emacs.d/elpa/elpy-1.12.0/`
(Windows) Python does not understand this path and can't find the elpy module when it starts
So I changed `elpy-rpc-pythonpath`:
``(setq elpy-rpc-pythonpath C:/windowspathtoemacs/.emacs.d/elpa/elpy-1.12.0/)``.
Now (I think) the elpy module is found (it runs and documentation works) but I get an error with rope and pyflakes
``flymake: Configuration error has occurred while running (pyflakes ). Flymake will be switched OFF``
``error in process filter: Elpy error: rope does not support files without a local project root``

Also I can't send anything to the console. I think some temp file is being opened and run but again because of this path issue Python (windows) can not find it. `C-c C-c` gets this error:
``FileNotFoundError: [Errno 2] No such file or directory: '/tmp/py11800DUv'``
Maybe I could change the name of the temp directory to something windows understands?

I apologize for all the questions. Thanks for making elpy! It works great on my mac :)

**Comment by jorgenschaefer** [comment]
Hello, and thanks for the report! This looks like a real mess there … Emacs thinks it's running on a Unix and uses Unix directory and file names, while Python thinks it's running on a Windows and gets very confused about the Unix directory names.

I'm afraid it might very well be impossible to fix this – you have to either use a non-Cygwin Emacs, or a cygwin Python.

(Caveat: I do not use Windows nor do I have any experience at all with Cygwin, so I could be very off, but that's what seems to be happening from the errors you posted.)

**Comment by benrifkind** [comment]
Yup. I think that is the problem. I guess I'll have to switch to Windows Emacs. Thanks for responding and appreciate the package.

**Comment by jorgenschaefer** [comment]
When you do, let me know if it works :-)

**Comment by benrifkind** [comment]
It works if I pass it a cygwin python distribution. I still can't pass code to the shell because I can't figure out how to change the interactive python to point to a cygwin python. The virtualenv points to the cygwin python path but that does not affect the interactive python path.

Virtualenv........: bin (/usr/bin)
RPC Python........: 3.4.3 (/usr/bin/python3)
Interactive Python: python (/cygdrive/c/windowspath/python)
Emacs.............: 24.5.1
Elpy..............: 1.12.0
Jedi..............: 0.9.0
Rope..............: 0.10.3
Importmagic.......: 0.1.7
Autopep8..........: Not found (1.2.4 available)
Yapf..............: 0.10.0
Syntax checker....: flake8 (/usr/bin/flake8)

Also, this may be a separate issue but code completion is extremely slow. I will attempt to code complete but nothing will show up for a few minutes

I'll let you know if a windows install of emacs works.

**Comment by MRFaria** [comment]
Don't know if you are still interested in a solution to this, but I use a windows emacs (mingw64) along with a windows python and get the rest of the unix tools like grep and find to work by adding the cygwin/msys2 packages to the emacs path explicitly.

`(if (eq system-type 'windows-nt)
    (progn (setq msys-bin "c:\\msys64\\usr\\bin")
	   (setq exec-path (cons msys-bin exec-path))
	   (setenv "PATH" (concat msys-bin ";" (getenv "PATH")))))`
